Modules/_collectionsmodule.c

```diff
@@ -1947,6 +2101,7 @@ static PyObject *
 deque_reviter(dequeobject *deque)
 {
     dequeiterobject *it;
+    Py_BEGIN_CRITICAL_SECTION(deque);
```

Comment:
This should be below the NULL check. The `return NULL` would return without ending the critical section.

Reply:
Good catch!